R/options_plumber.R

```diff
@@ -41,10 +41,12 @@
 #' \item{`plumber.legacyRedirects`}{Plumber will redirect legacy route `/__swagger__/` and
 #' `/__swagger__/index.html` to `../__docs__/` and `../__docs__/index.html`. You can disable this
 #' by settings this option to `FALSE`. Defaults to `TRUE`}
+#' \item{`plumber.typedParameters`}{Plumber will convert all parameters according to openApi
+#' supplied type hints.  Defaults to`TRUE`}
```

Comment:
One option here might be to default to FALSE - this would keep existing functionality intact... and the option could be changed to TRUE when Plumber is ready to make a bigger version release...

(although personally I want this set to TRUE in every API I write 👍 )

Reply:
If you are making a more major version change here... can I suggest we might also want to deprecate the typing from the dynamic routes - we don't want type information in both the param comments and in the dynamic routes?

Reply:
Yes. Let's go for it. But if `plumber.typedParameters` is not `TRUE`, then the old behavior should exist (if possible).

Reply:
I think that's what https://github.com/slodge/plumber/blob/cafc91ea24423306b19f0f71f750cb4b6ddf7aeb/R/plumber-step.R#L322 does

Reply:
Discussed with @schloerke in realtime, we think it makes sense to default to `FALSE` for now and then introduce the concept of editions, like testthat (https://testthat.r-lib.org/articles/third-edition.html).

@slodge, how do you feel about that?

Reply:
Happy for Posit/RStudio to choose how you version plumber and its features.

---------

I've not read about editions before, and I don't know if I understand right now how they'd work for a plumber app or for a helper package (e.g. how would the package specify that it relies on a plumber edition as well as its version?)

In this particular case, I suspect that you could get away with setting this to `TRUE` and incrementing the major plumber version (I also don't think this would actually break that many apps... but I've been wrong about that before!). 

But this is **100% the call for whoever maintains plumber** - aka **you** 👍